beacon-chain/sync/rpc_send_request.go

```diff
@@ -404,11 +405,8 @@ func readChunkedBlobSidecar(stream network.Stream, encoding encoder.NetworkEncod
 // SendDataColumnSidecarsByRangeRequest sends a request for data column sidecars by range
 // and returns the fetched data column sidecars.
 func SendDataColumnSidecarsByRangeRequest(
-	ctx context.Context,
-	tor blockchain.TemporalOracle,
-	p2pApi p2p.P2P,
+	p DataColumnSidecarsParams,
```

Comment:
It is awkward that this method needs a `DataColumnSidecarsParams` argument, but it does not use several fields from this value (`NewVerifier`, `Storage`).  I am currently refactoring backfill to call this method and I can't tell if this is a bug, or if it is by design. Maybe I can pass the params with the values set to nil, but that's a scary UX, because I don't know if this parameter is a footgun by design, or if there is a bug - in either case my code will panic if the values are ever used somewhere along this path.

Reply:
I don't want to block other peerdas work waiting on a redesign with Manu out next week, so I'm going to remove my change request and introduce some refactors to address the design issues with this PR in a new branch.